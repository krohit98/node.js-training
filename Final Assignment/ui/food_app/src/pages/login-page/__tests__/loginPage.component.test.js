import { render, screen } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import LoginPage from '../loginPage.component';
import * as reactRedux from "react-redux";
import reactRouter from "react-router";
import LoginForm from '../../../components/login-form/loginForm.component';

jest.mock('../../../components/login-form/loginForm.component');

describe("Unit testing LoginPage component",()=>{

    beforeEach(()=>{
        jest.spyOn(reactRedux, "useDispatch").mockReturnValue(jest.fn());
        jest.spyOn(reactRouter, "useLocation").mockReturnValue("abcd");
        jest.spyOn(reactRedux, "useSelector").mockReturnValue([]);
    })

    test("Snapshot testing",()=>{
        const renderer = new ShallowRenderer();
        renderer.render(<LoginPage />);
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    })

    test("Should render LoginPage correctly",()=>{
        LoginForm.mockImplementation(()=><div>LoginFormComponent</div>);
        render(<LoginPage />);
        const heading = screen.getByText("Sign-In to an Existing Account");
        const loginFormComponent = screen.getByText("LoginFormComponent");
        expect(heading).toBeInTheDocument();
        expect(loginFormComponent).toBeInTheDocument();
    })


})