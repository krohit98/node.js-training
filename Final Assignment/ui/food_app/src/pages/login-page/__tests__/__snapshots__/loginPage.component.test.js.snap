// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing LoginPage component Snapshot testing 1`] = `
<div
  className="pageDiv"
>
  <h1
    className="display-3 pageHeading"
  >
    Sign-In to an Existing Account
  </h1>
  <LoginForm
    submitHandler={[Function]}
  />
  
</div>
`;
